util/grub.d/00_menu_auto_hide.in

```diff
@@ -1,15 +1,18 @@
 #! /bin/sh
 
-# Disable / skip generating menu-auto-hide config parts on serial terminals
-for x in ${GRUB_TERMINAL_INPUT} ${GRUB_TERMINAL_OUTPUT}; do
-  case "$x" in
-    serial*)
-      exit 0
-      ;;
-  esac
-done
-
 cat << EOF
+# Boot Counting
+if [ "\${boot_success}" = "0"]; then
+  if  [ "\${boot_counter}" = "0" ]; then
+    set default=1
+    set fallback_active=true
+    save_env fallback_active
+  else
+    set boot_counter=$((\${boot_counter}-1))
+    save_env boot_counter
+  fi
+fi
+
```

Comment:
I don't think this belong to `00_menu_auto_hide.in`, since this an independent feature I would add it in a new script file (i.e: `01_fallback_support`).

That way we can for example ship the menu auto hide and fallback support independently depending on the Fedora variant that needs it, users can choose to disable one feature bit removing the script executable bit without affecting the other feature, etc.

Reply:
That way, both features couldn't be used at the same time. I'd like to have that option for workstations (even if it weren't active by default). I need the lines resetting and saving `boot_success` from this script. maybe @jwrdegoede has an idea how to separate things cleanly here?